src/passes/Precompute.cpp

```diff
@@ -39,106 +39,54 @@
 
 namespace wasm {
 
-static const Name NOTPRECOMPUTABLE_FLOW("Binaryen|notprecomputable");
-
 // Limit evaluation depth for 2 reasons: first, it is highly unlikely
 // that we can do anything useful to precompute a hugely nested expression
 // (we should succed at smaller parts of it first). Second, a low limit is
 // helpful to avoid platform differences in native stack sizes.
 static const Index MAX_DEPTH = 50;
 
+// Limit loop iterations since loops might be infinite.
+static const Index MAX_LOOP_ITERATIONS = 3;
+
 typedef std::unordered_map<LocalGet*, Literals> GetValues;
```

Comment:
please add a comment explaining how these get values differ from the parent class's.